### PR TITLE
feat: centralize module repo configuration

### PR DESCRIPTION
**Pull request by apeabody**

### CODE REVIEW

File: infra/terraform/test-org/org/locals.tf [hunk 1]
```diff
@@ -26,98 +26,393 @@ locals {
...
-    "cloud-foundation-training", # Not module
-    "cloud-router",
-    "group",
-    "data-fusion",
-    "module-template",
-    "vault",
-    "docs-samples",
-  ]
+  ci_folders       = module.folders-ci.ids
+  ci_repos_folders = { for repo in local.repos : try(repo.short_name, trimprefix(repo.name, "terraform-google-")) => { folder_name = "ci-${try(repo.short_name, trimprefix(repo.name, "terraform-google-"))}", folder_id = replace(module.folders-ci.ids["ci-${try(repo.short_name, trimprefix(repo.name, "terraform-google-"))}"], "folders/", ""), gh_org = repo.org } }
```
Comment (bharathkkb) [inline infra/terraform/test-org/org/locals.tf:30]: nit: it would be good to split this into multiple lines for readability

**Review by bharathkkb — APPROVED** [review]
Thanks @apeabody

File: infra/terraform/test-org/org/locals.tf (hunk 1 above)
Comment (apeabody) [reply infra/terraform/test-org/org/locals.tf:30]: SG

File: infra/terraform/modules/repositories/main.tf [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+resource "github_repository" "repo" {
+  for_each     = var.repos_map
+  name         = each.value.name
+  description  = try(each.value.description, null)
+  homepage_url = try(each.value.url, "https://registry.terraform.io/modules/${each.value.org}/${trimprefix(each.value.name, "terraform-google-")}/google")
+  topics       = setunion(["cft-terraform"], try(split(",", each.value.topics), []))
```
[the remark this answers is not in the document]
Comment (apeabody) [reply infra/terraform/modules/repositories/main.tf:22]: SG

**Comment by apeabody** [comment]
@bharathkkb applied

**Pull request merged**